File: src/main/java/com/ioannuwu/inline/ui/render/elements/MyGutterRenderer.java
package com.ioannuwu.inline.ui.render.elements;

import com.intellij.openapi.editor.markup.GutterIconRenderer;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.util.Objects;

public class MyGutterRenderer extends GutterIconRenderer {

    private final Icon icon;

    public MyGutterRenderer(@NotNull Icon icon) {
        this.icon = icon;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MyGutterRenderer that = (MyGutterRenderer) o;
        return icon.equals(that.icon);
    }

    @Override
    public int hashCode() {
        return Objects.hash(icon);
    }

    @Override
    public @NotNull Icon getIcon() {
        return icon;
    }
}

File: src/main/kotlin/com/ioannuwu/inline/domain/elements/MyGutterRenderer.kt
package com.ioannuwu.inline.domain.elements

import com.intellij.openapi.editor.markup.GutterIconRenderer
import javax.swing.Icon

class MyGutterRenderer(private val icon: Icon) : GutterIconRenderer() {

    override fun getIcon(): Icon = icon

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is GutterIconRenderer) return false
        return other.icon.hashCode() == hashCode()
    }

    override fun hashCode(): Int = icon.hashCode()
}